pkg/utils/backoff/backoff.go

```diff
@@ -0,0 +1,66 @@
+// Package backoff implements backoff algorithms for retrying operations.
```

Comment:
🤔 I'm not 100% wild about copying here to avoid the dependency -- it exists regardless of whether we copy it or not; the downside to the copy is that we now need to maintain it ourselves